tests/ut/test_simd.cc

```diff
@@ -13,203 +13,211 @@
+    auto x = std::make_unique<DataType[]>(rows * dim);
+    for (int i = 0; i < rows; i++) {
+        for (int j = 0; j < dim; j++) {
+            x[i * dim + j] = (DataType)distrib(rng);
+        }
+    }
     return x;
 }
 
-TEST_CASE("Test BruteForce Search SIMD", "[bf]") {
-    using Catch::Approx;
-
-    const int64_t nb = 1000;
-    const int64_t nq = 10;
-    const int64_t dim = 127;
-    const int64_t k = 5;
-
-    auto metric = GENERATE(as<std::string>{}, knowhere::metric::L2, knowhere::metric::COSINE);
-
-    const auto train_ds = GenDataSet(nb, dim);
-    const auto query_ds = CopyDataSet(train_ds, nq);
-
-    knowhere::Json conf = {
-        {knowhere::meta::DIM, dim},
-        {knowhere::meta::METRIC_TYPE, metric},
-        {knowhere::meta::TOPK, k},
-    };
-
-    auto test_search_with_simd = [&](knowhere::KnowhereConfig::SimdType simd_type) {
-        knowhere::KnowhereConfig::SetSimdType(simd_type);
-        auto gt = knowhere::BruteForce::Search<knowhere::fp32>(train_ds, query_ds, conf, nullptr);
-        REQUIRE(gt.has_value());
-        auto gt_ids = gt.value()->GetIds();
-        auto gt_dist = gt.value()->GetDistance();
-
-        for (int64_t i = 0; i < nq; i++) {
-            REQUIRE(gt_ids[i * k] == i);
-            if (metric == knowhere::metric::L2) {
-                REQUIRE(gt_dist[i * k] == 0);
-            } else {
-                REQUIRE(std::abs(gt_dist[i * k] - 1.0) < 0.00001);
-            }
+template <typename DataType>
+std::unique_ptr<DataType[]>
+ConvertVector(float* data, int dim, int rows) {
+    auto x = std::make_unique<DataType[]>(rows * dim);
+    for (int i = 0; i < rows; i++) {
+        for (int j = 0; j < dim; j++) {
+            x[i * dim + j] = (DataType)data[i * dim + j];
         }
-    };
-
-    for (auto simd_type : {knowhere::KnowhereConfig::SimdType::AVX512, knowhere::KnowhereConfig::SimdType::AVX2,
-                           knowhere::KnowhereConfig::SimdType::SSE4_2, knowhere::KnowhereConfig::SimdType::GENERIC,
-                           knowhere::KnowhereConfig::SimdType::AUTO}) {
-        test_search_with_simd(simd_type);
     }
+    return x;
 }
 
-TEST_CASE("Test PQ Search SIMD", "[pq]") {
+TEST_CASE("Test distance") {
     using Catch::Approx;
+    auto simd_type = GENERATE(as<knowhere::KnowhereConfig::SimdType>{}, knowhere::KnowhereConfig::SimdType::AVX512,
+                              knowhere::KnowhereConfig::SimdType::AVX2, knowhere::KnowhereConfig::SimdType::SSE4_2,
+                              knowhere::KnowhereConfig::SimdType::GENERIC, knowhere::KnowhereConfig::SimdType::AUTO);
+    auto dim = GENERATE(as<size_t>{}, 1, 7, 14, 21, 28, 35, 42, 49, 56, 64, 128, 256, 512);
+
+    LOG_KNOWHERE_INFO_ << "simd type: " << simd_type << ", dim: " << dim;
+    knowhere::KnowhereConfig::SetSimdType(simd_type);
+
+    SECTION("test single distance calculation") {
+        const size_t nx = 1, ny = 1;
+
+        const float tolerance = 0.000001;
```

Comment:
Please use `0.000001f` in order to avoid compiler warnings in a log.
The same note for other fp numbers down below.

Reply:
OK